// nexmo/lib/FilesResource.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

var fs = require('fs');

var FilesResource = function () {
  _createClass(FilesResource, null, [{
    key: 'PATH',


    /**
     * The path to the `calls` resource.
     */
    get: function get() {
      return '/v1/files';
    }

    /**
     * Creates a new FilesResource.
     *
     * @param {Credentials} creds - Credentials used when interacting with the Nexmo API.
     * @param {Object} options - additional options for the class.
     */

  }]);

  function FilesResource(creds, options) {
    _classCallCheck(this, FilesResource);

    this.creds = creds;
    this.options = options;
  }

  /**
   * Get stream for a remote File
   *
   * @param {string} [fileIdOrUrl] - The unique identifier or URL for the file
   * @param {function} callback - function to be called when the request completes.
   */


  _createClass(FilesResource, [{
    key: 'get',
    value: function get(fileIdOrUrl, callback) {

      if (!fileIdOrUrl) {
        throw new Error('"fileIdOrUrl" is a required parameter');
      }

      fileIdOrUrl = fileIdOrUrl.split("/").pop(-1);

      var config = {
        host: 'api.nexmo.com',
        path: FilesResource.PATH + '/' + fileIdOrUrl,
        method: 'GET',
        headers: {
          'Content-Type': 'application/octet-stream',
          'Authorization': 'Bearer ' + this.creds.generateJwt()
        }
      };

      this.options.httpClient.request(config, callback);
    }

    /**
     * Save remote File locally
     *
     * @param {string} [fileIdOrUrl] - The unique identifier or URL for the file
     * @param {string} [file] - Filename or file descriptor
     * @param {function} callback - function to be called when the request completes.
     */

  }, {
    key: 'save',
    value: function save(fileIdOrUrl, file, callback) {
      var _this = this;

      this.get(fileIdOrUrl, function (error, data) {
        if (error) {
          callback(error, null);
        } else {
          _this.__storeFile(data, file, callback);
        }
      });
    }
  }, {
    key: '__storeFile',
    value: function __storeFile(data, file, callback) {
      fs.writeFile(file, data, function (error) {
        if (error) {
          callback(error, null);
        } else {
          callback(null, file);
        }
      });
    }
  }]);

  return FilesResource;
}();

exports.default = FilesResource;
module.exports = exports['default'];
